feat(jaeger-operator): upgrade Jaeger Operator from v1.22 with breaking changes (#46)
sighupio/fury-kubernetes-service-mesh: katalog/istio-operator/jaeger-operator/BREAKING.md

## Comprehensive Upgrade Guide for the Jaeger Operator from v1.22 to the Latest Release 🦮

### Introduction

> This guide provides detailed instructions for upgrading the Jaeger Operator component in a Service Mesh from version 1.22 to the latest release. The previous version is substantially outdated, and a direct upgrade path is unavailable. As a result, it is necessary to completely remove the old version before installing the new one. This document explains the reasons for this approach and outlines the steps to successfully transition to the latest version, which can be explored in detail [here](https://www.jaegertracing.io/docs/1.57/operator/).

Reviewer: It would be better to specify to which specific version instead of saying just "latest".

Also, the link points to 1.57 but the manifests to 1.56, which one should be?

```suggestion
This guide provides detailed instructions for upgrading the Jaeger Operator component in a Service Mesh from version 1.22 to the latest release at the moment (1.56). The previous version is substantially outdated, and a direct upgrade path is unavailable. As a result, it is necessary to completely remove the old version before installing the new one. This document explains the reasons for this approach and outlines the steps to successfully transition to the latest version, which can be explored in detail [here](https://www.jaegertracing.io/docs/1.56/operator/).
```

### Breaking Changes Notification 💔

#### Key Points
- **Version Obsolescence:** The Jaeger Operator version 1.22 is no longer supported, and direct updates to the new version are not possible.

> - **Impact on Cluster Operations:** The removal of the old version and installation of the new one may temporarily affect the stability and functionality of your Service Mesh.

Reviewer: out of curiosity, why may removing Jeager operator affect the service mesh? Isn't it used just for visualization?

Author: @stefanoghinelli could provide a better answer, but according to the procedure, the previous version needs to be removed in order to install the new one, which could impact the operations in the cluster. The Jeager Operator and Operated would be directly affected. This is a breaking change because some custom implementations related to Jaeger could be affected, and we don't have any direct upgrade path for this. As for the impact on the service mesh, I'm not entirely sure; perhaps @stefanoghinelli  can provide more details about it.

#### Actions Required
- **Complete Resource Deletion:** All resources associated with the old version must be removed to prevent conflicts or lingering configurations that could impact the new installation.
- **Manual Reinstallation Needed:** After clearing the old resources, a fresh installation of the new version is required.

### Upgrade Guide 🦮

#### Preparing for the Upgrade

> #### Current Configuration Snapshot
> - **Perform a kustomize build:** Run `kustomize build .` on the current Jaeger Operator deployment and save the output. This step is crucial for identifying all the resources currently deployed under the old version. This information will be used to ensure a complete cleanup before installing the new version.

Reviewer: An empty line should be left after a title, Two lines in markdown are merged if there's no an empty line between them.

```suggestion
#### Current Configuration Snapshot

- **Perform a kustomize build:** Run `kustomize build .` on the current Jaeger Operator deployment and save the output. This step is crucial for identifying all the resources currently deployed under the old version. This information will be used to ensure a complete cleanup before installing the new version.
```

> #### Comprehensive Resource Removal
> - **Remove All Previous Resources:** To guarantee a successful update, it is essential to delete all resources associated with the previous version of the Jaeger Operator. This includes deployments, service accounts, roles, role bindings, and any custom resources defined by Jaeger. Removing these resources prevents any potential conflicts or leftover configurations that could interfere with the new version's operations.

Reviewer: ```suggestion
#### Comprehensive Resource Removal

- **Remove All Previous Resources:** To guarantee a successful update, it is essential to delete all resources associated with the previous version of the Jaeger Operator. This includes deployments, service accounts, roles, role bindings, and any custom resources defined by Jaeger. Removing these resources prevents any potential conflicts or leftover configurations that could interfere with the new version's operations.
```

> #### Backup and Risk Management

Reviewer: The backup step should be before the removal IMHO

> #### Backup and Risk Management
> - **Backup Critical Data and Configurations:** Before proceeding with the update, make sure to backup all important configurations and data related to Jaeger. This precaution helps to safeguard your data in the event of an issue during the upgrade process.

Reviewer: ```suggestion
#### Backup and Risk Management

- **Backup Critical Data and Configurations:** Before proceeding with the update, make sure to backup all important configurations and data related to Jaeger. This precaution helps to safeguard your data in the event of an issue during the upgrade process.
```

- **Prepare for Potential Disruptions:** Be aware that removing the Jaeger Operator and its associated resources may temporarily impact the services depending on it. Plan for possible service interruptions during this transition phase and inform relevant stakeholders of the expected downtime.

#### Installing the New Version

> Following the cleanup of the old resources, proceed with the new version installation. Refer to the latest Jaeger Operator documentation provided [here](https://www.jaegertracing.io/docs/1.57/operator/) for detailed installation instructions. Ensure that you follow these guidelines closely to integrate the new version correctly into your environment.

Reviewer: ```suggestion
Following the cleanup of the old resources, proceed with the new version installation. Refer to the latest Jaeger Operator documentation provided [here](https://www.jaegertracing.io/docs/1.56/operator/) for detailed installation instructions. Ensure that you follow these guidelines closely to integrate the new version correctly into your environment.
```

### Conclusion

Upgrading to the latest version of the Jaeger Operator requires careful planning and execution due to the significant changes involved. By following this guide, you can ensure a smooth transition and minimize potential disruptions to your Service Mesh operations. Always be prepared for operational impacts and have appropriate recovery strategies in place.

> ### Important Notice
> - **No Migration Path:** There is no migration path available from Jaeger Operator version 1.22 to the latest version. The only option is to remove the old version entirely and then install the new operator.

Reviewer: ```suggestion
### Important Notice

- **No Migration Path:** There is no migration path available from Jaeger Operator version 1.22 to the latest version. The only option is to remove the old version entirely and then install the new operator.
```

> - **Namespace Verification:** Ensure to verify the current namespace being used in your cluster as the namespace might change. This verification helps in ensuring a smooth cleanup and installation process.

Reviewer: this item needs a little rewording, I think I get the idea but it is not clear.
